apps/openchallenges/infra/src/openchallenges-preview-stack.ts

```diff
@@ -106,6 +107,9 @@ export class OpenChallengesPreviewStack extends SageStack {
       previewInstance.instance.privateIp
     );
 
+    // The DNS configuration
+    const dns = new Dns(this, 'dns', previewInstanceAlb.lb);
```

Comment:
Is this going to rebuild the Dns construct every time (and tear it down if you tear down the app app stack)? I think we should move the creation of the Dns record 'preview.openchallenges.io' in this stack (and construct the DNS zone in another stack and reference it here). The zone needs to be independent of the app stack because we'll need to delegate DNS to it in the root zone (openchallenges.io in sageit).

Reply:
I created another stack named `cnb-dev` to deploy "permanent" and shared resources like zones and certs. See more info about this new stack below.

Reply:
The dev DNS zone and certificate are now created as part of a separate stack named `cnb-dev` (see below).